bye "discord.js"

Mogyuchi opened the pull request:
ライブラリ変更に伴う変更点がわかりやすいようにあえてカスな実装にしている箇所が多数あります。

Mogyuchi committed:
bye "discord.js"

Mogyuchi commented:
あ、だめそう

cm-ayf reviewed (commented):
全部見たので、ここからカスな実装をそうでない実装にしていただくと、ここからのdiffをレビューできる

Mogyuchi commented:
別プロジェクトを優先するため、保留します。

そちら側で修正していただいても構いません。

cm-ayf commented on main.js:
```diff
@@ -1,34 +1,80 @@
+client.on(GatewayDispatchEvents.Ready, async ({ data: readyData }) => {
+  console.log(
+    `Logged in as ${
+      readyData.user?.discriminator === "0"
+        ? readyData.user?.username
+        : `${readyData.user?.username}#${readyData.user?.discriminator}`
```
ここはもうusernameにして良さそう

Mogyuchi replied:
Bot は新ユーザー名が未適用なので username が一意じゃないけどいいの

cm-ayf replied:
まあいいでしょう　自分の持ってるトークンのうちどれであるかを確かめる程度にしか使わないと思うし

cm-ayf commented on main.js:
```diff
@@ -1,34 +1,80 @@
+      allowed_mentions: { replied_user: false },
+    };
+    for (const user of bellUsers) {
+      await Promise.race([
+        api.channels.createMessage(
+          (
+            await api.users.createDM(user.id)
+          ).id,
+          options
+        ),
+        setTimeout(1000),
+      ]);
```
すみません，これPromise.allにしてもらっていいですか……？

cm-ayf commented on main.js:
```diff
@@ -1,34 +1,80 @@
+      content: `**${by}** reacted to https://discord.com/channels/${
+        reactionAddData.guild_id ?? "@me"
+      }/${reactionAddData.channel_id}/${reactionAddData.message_id} with ${
+        reactionAddData.emoji.id
+          ? `<${reactionAddData.emoji.animated ? "a" : ""}:${
+              reactionAddData.emoji.name
+            }:${reactionAddData.emoji.id}>`
+          : reactionAddData.emoji.name
```
関数に切り出してください．

cm-ayf commented on main.js:
```diff
@@ -1,34 +1,80 @@
+    if (!bellUsers.length) return;
+
+    const by =
+      reactionAddData.member?.nick ??
+      reactionAddData.member?.user?.global_name ??
+      reactionAddData.member?.user?.username;
```
これも関数に切り出した方がいいのかなぁ

Mogyuchi replied:
てかこれ、user.tag に準拠してないやん()

Mogyuchi replied:
そもそもメンションにすべき、というお話になった

Mogyuchi committed:
update

- ログインユーザーの提示に username を用いるよう変更
- リアクションしたユーザーをメンションで示すように変更
- 絵文字文字列の取得を関数に切り出し
- DM送信を Promise.all に変更
